sdk/state/ingest.go

```diff
@@ -11,8 +11,19 @@ import (
 // unsuccessful on the network. The function updates the internal state of the
 // channel if the transaction relates to the channel.
 func (c *Channel) IngestTx(txXDR, resultXDR, resultMetaXDR string) error {
-	// TODO: Use the transaction result to affect on success/failure.
+	// If channel has not been opened or has been closed, return.
+	if c.OpenAgreement().isEmpty() {
+		return fmt.Errorf("channel has not been opened")
+	}
+	cs, err := c.State()
+	if err != nil {
+		return fmt.Errorf("getting channel state: %w", err)
+	}
+	if cs == StateClosed {
+		return fmt.Errorf("channel has been closed")
```

Comment:
fyi: the ticket says to end early but if the channel isn't opened I think it should be a user error, since they're trying to ingest at a time they shouldn't be and should be told why

Reply:
I'm curious if this error causes the console example to error unexpectedly and therefore if we need to handle errors differently further up the stack. The reason I'm wondering this is because ingestion happens on another go routine I think we will call IngestTx when we're not supposed to. It might be all fine.

It doesn't have to occur in this PR, but I think there could be some value in us having a sentinel error (var ErrChannelClosed = errors.New("channel closed")` or an error type `type ErrChannelClosed` that we return in all places that the channel is being accessed in a way it can't be. The error could be `ErrChannelNotOpen` so it could be used pre-open and post-close.